Sources/NonEmpty/NonEmpty+String.swift

```diff
@@ -1,24 +1,130 @@
 public typealias NonEmptyString = NonEmpty<String>
 
-extension NonEmpty where C == String {
-  public init(_ head: Character) {
-    self.init(head, "")
+extension NonEmpty: ExpressibleByUnicodeScalarLiteral where C: ExpressibleByUnicodeScalarLiteral {
+  public typealias UnicodeScalarLiteralType = C.UnicodeScalarLiteralType
+
+  public init(unicodeScalarLiteral value: C.UnicodeScalarLiteralType) {
+    self.init(rawValue: C(unicodeScalarLiteral: value))!
+  }
+}
+
+extension NonEmpty: ExpressibleByExtendedGraphemeClusterLiteral where C: ExpressibleByExtendedGraphemeClusterLiteral {
+  public typealias ExtendedGraphemeClusterLiteralType = C.ExtendedGraphemeClusterLiteralType
+
+  public init(extendedGraphemeClusterLiteral value: C.ExtendedGraphemeClusterLiteralType) {
+    self.init(rawValue: C(extendedGraphemeClusterLiteral: value))!
+  }
+}
+
+extension NonEmpty: ExpressibleByStringLiteral where C: ExpressibleByStringLiteral {
+  public typealias StringLiteralType = C.StringLiteralType
+
+  public init(stringLiteral value: C.StringLiteralType) {
+    self.init(rawValue: C(stringLiteral: value))!
+  }
+}
+
+extension NonEmpty: TextOutputStreamable where C: TextOutputStreamable {
+  public func write<Target>(to target: inout Target) where Target : TextOutputStream {
+    self.rawValue.write(to: &target)
+  }
+}
+
+extension NonEmpty: TextOutputStream where C: TextOutputStream {
+  public mutating func write(_ string: String) {
+    self.rawValue.write(string)
+  }
+}
+
+extension NonEmpty: LosslessStringConvertible where C: LosslessStringConvertible {
+  public init?(_ description: String) {
+    guard let string = C(description) else { return nil }
+    self.init(rawValue: string)
+  }
+}
+
+extension NonEmpty: ExpressibleByStringInterpolation where C: ExpressibleByStringInterpolation, C.StringLiteralType == DefaultStringInterpolation.StringLiteralType {}
+
+#if swift(>=5.2)
+extension NonEmpty: StringProtocol where C: StringProtocol {
```

Comment:
Also added conformance to `StringProtocol`, which adds a bunch of functionality and was useful in Point-Free for using literals in tests. We could potentially add literal code for arrays/dictionaries, but we'd have to do that unsafe bit-cast dance that Joe told us not to do with Tagged (Slava thinks it's fine with ABI stability though).